src/libsync/updatefiledropmetadata.cpp

```diff
@@ -0,0 +1,193 @@
+{
+    qCDebug(lcUpdateFileDropMetadataJob) << "Metadata Received, Preparing it for the new file." << json.toVariant();
+
+    // Encrypt File!
+    _metadata = new FolderMetadata(propagator()->account(), json.toJson(QJsonDocument::Compact), statusCode);
+    if (!_metadata->moveFileDropToFiles()) {
+        unlockFolder();
+        return;
+    }
+
+    const auto updateMetadataJob = new UpdateMetadataApiJob(propagator()->account(), _folderId, _metadata->encryptedMetadata(), _folderToken);
+    connect(updateMetadataJob, &UpdateMetadataApiJob::success, this, &UpdateFileDropMetadataJob::slotUpdateMetadataSuccess);
+    connect(updateMetadataJob, &UpdateMetadataApiJob::error, this, &UpdateFileDropMetadataJob::slotUpdateMetadataError);
+    updateMetadataJob->start();
+}
+
+void UpdateFileDropMetadataJob::slotUpdateMetadataSuccess(const QByteArray &fileId)
+{
+    Q_UNUSED(fileId);
+    qCDebug(lcUpdateFileDropMetadataJob) << "Uploading of the metadata success, Encrypting the file";
+
+    qCDebug(lcUpdateFileDropMetadataJob) << "Finalizing the upload part, now the actuall uploader will take over";
+    unlockFolder();
+}
+
+void UpdateFileDropMetadataJob::slotUpdateMetadataError(const QByteArray &fileId, int httpErrorResponse)
+{
+    qCDebug(lcUpdateFileDropMetadataJob) << "Update metadata error for folder" << fileId << "with error" << httpErrorResponse;
+    qCDebug(lcUpdateFileDropMetadataJob()) << "Unlocking the folder.";
+    unlockFolder();
+}
+
+void UpdateFileDropMetadataJob::slotFolderLockedError(const QByteArray &fileId, int httpErrorCode)
+{
+    Q_UNUSED(httpErrorCode);
+    qCDebug(lcUpdateFileDropMetadataJob) << "Folder" << fileId << "Coundn't be locked.";
+    emit finished(SyncFileItem::Status::NormalError);
+}
+
+void UpdateFileDropMetadataJob::slotFolderEncryptedIdError(QNetworkReply *r)
+{
+    Q_UNUSED(r);
+    qCDebug(lcUpdateFileDropMetadataJob) << "Error retrieving the Id of the encrypted folder.";
+}
+
+void UpdateFileDropMetadataJob::unlockFolder()
+{
+    ASSERT(!_isUnlockRunning);
+
+    if (!_isFolderLocked) {
+        emit finished(SyncFileItem::Status::Success);
+        return;
+    }
+
+    if (_isUnlockRunning) {
+        qWarning() << "Double-call to unlockFolder.";
+        return;
+    }
+
+    _isUnlockRunning = true;
+
+    qDebug() << "Calling Unlock";
+    const auto unlockJob = new UnlockEncryptFolderApiJob(propagator()->account(), _folderId, _folderToken, this);
+
+    connect(unlockJob, &UnlockEncryptFolderApiJob::success, [this](const QByteArray &folderId) {
+        qDebug() << "Successfully Unlocked";
+        _folderToken = "";
+        _folderId = "";
+        _isFolderLocked = false;
+
+        emit folderUnlocked(folderId, 200);
+        _isUnlockRunning = false;
+        emit finished(SyncFileItem::Status::Success);
+    });
+    connect(unlockJob, &UnlockEncryptFolderApiJob::error, [this](const QByteArray &folderId, int httpStatus) {
+        qDebug() << "Unlock Error";
```

Comment:
Seems like these `qDebug` and `qWarning` should be changed for `qCDebug` and `qCWarning`